sw/device/lib/crypto/drivers/hmac.c

```diff
@@ -77,28 +78,42 @@ enum {
  *
  * TODO(#23191): It might be beneficial to have a timeout value for the polling.
  *
- * @return Result of the operation.
+ * @param[out] status of the operation.
+ * @param[out] ctx Context to which values are written.
+ * @param[out] hw_recovered If HW has been recovered after a hang.
  */
-OT_WARN_UNUSED_RESULT
-static status_t hmac_idle_wait(void) {
+static void hmac_idle_wait(
+  status_t *status, hmac_ctx_t *ctx, bool *hw_recovered) {
   // Verify that HMAC HWIP is idle.
   // Initialize `status_reg = 0` so that the loop starts with the assumption
   // that HMAC HWIP is not idle.
+  // TODO: should be removed when issue #24767 will be solved in the HW
+  //  | At max 4 clock cycles are required to perform a read access to the
+  //  | register. As it should take less than 64 clock cycles in SHA2-256 and
+  //  | 80 clock cycles in SHA2-384/512, let's take some margin and consider
+  //  | that 50 loops are a way enough to see IDLE status. Otherwise we can
+  //  | start to attempt to recover the HW.
   uint32_t status_reg = 0;
+  uint32_t attempt_cnt = 0;
   while (bitfield_bit32_read(status_reg, HMAC_STATUS_HMAC_IDLE_BIT) == 0) {
     status_reg = abs_mmio_read32(kHmacBaseAddr + HMAC_STATUS_REG_OFFSET);
+    attempt_cnt++;
+    if (attempt_cnt == 50) {
+      recover_hw_after_stop(ctx);
+      hw_recovered = true;
```

Comment:
To write under the pointer, you might need:
```suggestion
      *hw_recovered = true;
```
Might be what's causing the compilation issue?